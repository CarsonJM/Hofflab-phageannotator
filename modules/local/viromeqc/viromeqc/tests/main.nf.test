nextflow_process {

    name "Test Process: VIROMEQC_VIROMEQC"
    script "../main.nf"
    process "VIROMEQC_VIROMEQC"

    setup {
        run("VIROMEQC_INSTALL") {
            script "../../install/main.nf"
        }
    }

    test("fastq.gz single") {
        when {
            process {
                """
                input[0] = [
                    [ id:'test' ], // meta map
                    [
                        file(params.modules_testdata_base_path + 'genomics/sarscov2/illumina/fastq/test_1.fastq.gz', checkifExists: true),
                    ]
                ]
                input[1] = VIROMEQC_INSTALL.out.viromeqc_index
                """
            }
        }

        then {
            assertAll (
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }

    test("fastq.gz paired") {
        when {
            process {
                """
                input[0] = [
                    [ id:'test' ], // meta map
                    [
                        file(params.modules_testdata_base_path + 'genomics/sarscov2/illumina/fastq/test_1.fastq.gz', checkifExists: true),
                        file(params.modules_testdata_base_path + 'genomics/sarscov2/illumina/fastq/test_2.fastq.gz', checkifExists: true),
                    ]
                ]
                input[1] = VIROMEQC_INSTALL.out.viromeqc_index
                """
            }
        }

        then {
            assertAll (
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }

    test("fastq.gz paired - stub") {

        options "-stub"

        when {
            process {
                """
                input[0] = [
                    [ id:'test' ], // meta map
                    [
                        file(params.modules_testdata_base_path + 'genomics/sarscov2/illumina/fastq/test_1.fastq.gz', checkifExists: true),
                        file(params.modules_testdata_base_path + 'genomics/sarscov2/illumina/fastq/test_2.fastq.gz', checkifExists: true),
                    ]
                ]
                input[1] = VIROMEQC_INSTALL.out.viromeqc_index
                """
            }
        }

        then {
            assertAll (
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }
}
