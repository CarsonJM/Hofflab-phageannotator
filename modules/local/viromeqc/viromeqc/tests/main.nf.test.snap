{
    "fastq.gz single": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test.viromeqc.tsv:md5,04c6731f1848980468ff11f9286e5f1f"
                    ]
                ],
                "1": [
                    "versions.yml:md5,04c1eaabeaaa45359f8cebd60f7ade20"
                ],
                "enrichment": [
                    [
                        {
                            "id": "test"
                        },
                        "test.viromeqc.tsv:md5,04c6731f1848980468ff11f9286e5f1f"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,04c1eaabeaaa45359f8cebd60f7ade20"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.04.1"
        },
        "timestamp": "2024-07-23T09:29:38.361341306"
    },
    "fastq.gz paired": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test.viromeqc.tsv:md5,6acbffc97f75774cf9fcc953b98913ed"
                    ]
                ],
                "1": [
                    "versions.yml:md5,04c1eaabeaaa45359f8cebd60f7ade20"
                ],
                "enrichment": [
                    [
                        {
                            "id": "test"
                        },
                        "test.viromeqc.tsv:md5,6acbffc97f75774cf9fcc953b98913ed"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,04c1eaabeaaa45359f8cebd60f7ade20"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.04.1"
        },
        "timestamp": "2024-07-23T09:32:29.89726697"
    },
    "fastq.gz paired - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test.viromeqc.tsv:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    "versions.yml:md5,04c1eaabeaaa45359f8cebd60f7ade20"
                ],
                "enrichment": [
                    [
                        {
                            "id": "test"
                        },
                        "test.viromeqc.tsv:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,04c1eaabeaaa45359f8cebd60f7ade20"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.04.1"
        },
        "timestamp": "2024-07-23T09:42:35.059686334"
    }
}